src/main/java/org/ibp/api/java/impl/middleware/common/validator/GermplasmUpdateValidator.java

```diff
@@ -0,0 +1,171 @@
+package org.ibp.api.java.impl.middleware.common.validator;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.generationcp.commons.util.DateUtil;
+import org.generationcp.middleware.api.attribute.AttributeDTO;
+import org.generationcp.middleware.api.breedingmethod.BreedingMethodService;
+import org.generationcp.middleware.api.germplasm.GermplasmNameTypeDTO;
+import org.generationcp.middleware.domain.germplasm.GermplasmUpdateDTO;
+import org.generationcp.middleware.manager.api.GermplasmDataManager;
+import org.generationcp.middleware.manager.api.LocationDataManager;
+import org.generationcp.middleware.pojos.Germplasm;
+import org.generationcp.middleware.pojos.Location;
+import org.ibp.api.java.germplasm.GermplasmService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.BindingResult;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class GermplasmUpdateValidator {
```

Comment:
General comment to standarize our code
I am considering the keys of the map case insensitive. For example, if an user use DRVNAME or drvname to me is the same. 
But given that when we get values for the map the comparisson is made by equals I am using toUpperCase in many places of the code 
Not sure if you handle this case and what is the expected result if an user enter twice the same attribute or name? Example "drvname", "DRVNAME"? Trying to update the same name type twice ? 
In my case I just reject the request because the key is duplicated.
The same with the preferredName, if the keyName is upper case and preferred name is lower case it is still valid to me. I Can see this onconsistency:

This is valid: 
```
    "names": {
      "LNAME": "NewLName",
      "accname": "New preferred name"
    },
    "preferredName": "ACCNAME",
```

But this is not: 
```
    "names": {
      "LNAME": "NewLName",
      "accname": "New preferred name"
    },
    "preferredName": "accname",
```

Reply:
I'll update the code make the fcode case insensitive as part of IBP-4154.